src/Microsoft.Azure.SignalR.Common/ServiceConnections/ManagementServiceConnectionContainer.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.SignalR.Common.ServiceConnections
+{
+    internal class ManagementServiceConnectionContainer : WeakServiceConnectionContainer
+    {
+        private bool _manuallyStop;
+
+        public ManagementServiceConnectionContainer(IServiceConnectionFactory serviceConnectionFactory,
+            int fixedConnectionCount, HubServiceEndpoint endpoint, ILogger logger = null) : base(
+            serviceConnectionFactory, fixedConnectionCount, endpoint, logger)
+        {
+            _manuallyStop = false;
+        }
+
+        protected override async Task StartCoreAsync(IServiceConnection connection, string target = null)
+        {
+            try
+            {
+                await connection.StartAsync(target);
+            }
+            finally
+            {
+                if (!_manuallyStop)
+                {
+                    await OnConnectionComplete(connection);
+                }
+            }
+        }
+
+        public override Task StopAsync()
```

Comment:
Shall we rethink our current design? I do feel with explicit StopAsync, the container, no matter which, should stop the connection. Cc @zackliu

Reply:
Shall we move the reconnect part to connection level instead of container level? Single connection also has this issue.

Add a bool to mark the explicit stop? Or throw a special exception to skip reconnect?